src/box/box.cc

```diff
@@ -4310,11 +4310,21 @@ box_process_fetch_snapshot(struct iostream *io,
 			  "wal_mode = 'none'");
 	}
 
-	say_info("sending current read-view to replica at %s", sio_socketname(io->fd));
+	say_info("sending read-view to replica at %s", sio_socketname(io->fd));
+	/* Used for checkpoint initial join. */
+	struct checkpoint_cursor cursor;
+	struct checkpoint_cursor *cursor_ptr = NULL;
+	if (req.is_checkpoint_join) {
+		cursor.vclock = &req.checkpoint_vclock;
+		cursor.start_lsn = req.checkpoint_lsn;
+		cursor.lsn_counter = 0;
+		cursor_ptr = &cursor;
+	}
```

Comment:
Please, either assign the struct via `var = {.field1 = ..., .field2 = .... }` or make a `memset(0)` before filling it. Otherwise we have a high risk to get corrupted data here when the struct will grow in the future.

Reply:
```
/home/serpentian/Programming/tnt/tarantool-ee/tarantool/src/box/box.cc: In function ‘void box_process_fetch_snapshot(iostream*, const xrow_header*)’:
/home/serpentian/Programming/tnt/tarantool-ee/tarantool/src/box/box.cc:4321:42: error: narrowing conversion of ‘req.fetch_snapshot_request::checkpoint_lsn’ from ‘uint64_t’ {aka ‘long unsigned int’} to ‘int64_t’ {aka ‘long int’} [-Werror=narrowing]
 4321 |                         .start_lsn = req.checkpoint_lsn,
      |                                      ~~~~^~~~~~~~~~~~~~
At global scope:
```